aruco python binding: refineDetectedMarkers

opencv-modules commithash: ba1d3ef99cf5e67241c5a31dbd9c344d12a2f7c5
opencv commithash: f6fa1cee2b9fc726452779cfd484ac382c142bab
Windows 8 x64
vc14 x64 toolchain Release mode
python 27 x64
numpy==1.9.3 x64

Code below causes exception 
```
more_corners, more_ids, rej, recovered_ids = cv2.aruco.refineDetectedMarkers(img, board, corners, ids, rejected)
TypeError: rejectedCorners is not a numpy array, neither a scalar

```
```
import cv2
import numpy as np

tag_size_m = 0.058
tag_margin_m = 0.012
board = cv2.aruco.GridBoard_create(2, 4, tag_size_m, tag_margin_m, cv2.aruco.getPredefinedDictionary(4))

cap = cv2.VideoCapture(0)

while True:
    _, img = cap.read()

    corners, ids, rejected = cv2.aruco.detectMarkers(img, board.dictionary)

    if 0 < len(corners) < len(board.objPoints):
        more_corners, more_ids, rej, recovered_ids = cv2.aruco.refineDetectedMarkers(img, board, corners, ids, rejected)

    k = cv2.waitKey(1) & 0xff
    if k == 27:
        break

```

If I reassign `rejected ` as `rejected = np.array(rejected)` it throws exception:

```
Traceback (most recent call last):
    more_corners, more_ids, rej, recovered_ids = cv2.aruco.refineDetectedMarkers(img, board, corners, ids, rejected)
cv2.error: ......\opencv\modules\core\src\matrix.cpp:553: error: (-215) r == Range::all() || (0 <= r.start && r.start < r.end && r.end <= m.size[i]) in function cv::Mat::Mat
```
shape of rejected in that case is (13L, 1L, 4L, 2L)

Any of my attempts to fix it by arrays ended with same exception. I've also tried: 
`rejected = np.reshape(np.array(rejected), (len(rejected) * 4, 2))`
`rejected = np.array(rejected).squeeze()`



Structure of `rejected` is similar with structure of `corners`.
But in c++ they have different type in signature of refineDetectedMarkers, is it ok?
```
void refineDetectedMarkers(InputArray _image, Ptr<Board> &_board,
                           InputOutputArrayOfArrays _detectedCorners, InputOutputArray _detectedIds,
                           InputOutputArray _rejectedCorners, InputArray _cameraMatrix,
                           InputArray _distCoeffs, float minRepDistance, float errorCorrectionRate,
                           bool checkAllOrders, OutputArray _recoveredIdxs,
                           const Ptr<DetectorParameters> &_params)
```

How I can handle that?

I've tried that:
```
    if 0 < len(corners) < len(board.objPoints):
        for rejected in rejected:
            more_corners, more_ids, rej, recovered_ids = cv2.aruco.refineDetectedMarkers(img, board, corners, ids, rejected)
```
There is exaclty the same exception: 
`cv2.error: ......\opencv\modules\core\src\matrix.cpp:553: error: (-215) r == Range::all() || (0 <= r.start && r.start < r.end && r.end <= m.size[i]) in function cv::Mat::Mat`

Line changed because openCV was pulled and rebuilded.

Moreover, in refineDetectedMarkers c++ code there exist line:
`Point2f rejCorner = _rejectedCorners.getMat(j).ptr< Point2f >()[(c + k) % 4];`
that tells that expected shape of rejectedCorners is (n,4,2), isn't it?
https://github.com/Itseez/opencv_contrib/blob/master/modules/aruco/src/aruco.cpp#L1168

--- a/modules/aruco/include/opencv2/aruco.hpp
+++ b/modules/aruco/include/opencv2/aruco.hpp
@@ -374,7 +374,7 @@ CV_EXPORTS_W int estimatePoseBoard(InputArrayOfArrays corners, InputArray ids, P
  */
 CV_EXPORTS_W void refineDetectedMarkers(
     InputArray image, Ptr<Board> &board, InputOutputArrayOfArrays detectedCorners,
-    InputOutputArray detectedIds, InputOutputArray rejectedCorners,
+    InputOutputArray detectedIds, InputOutputArrayOfArrays rejectedCorners,
     InputArray cameraMatrix = noArray(), InputArray distCoeffs = noArray(),
     float minRepDistance = 10.f, float errorCorrectionRate = 3.f, bool checkAllOrders = true,
     OutputArray recoveredIdxs = noArray(), const Ptr<DetectorParameters> &parameters = DetectorParameters::create());

--- a/modules/aruco/src/aruco.cpp
+++ b/modules/aruco/src/aruco.cpp
@@ -1099,7 +1099,7 @@ static void _projectUndetectedMarkers(Ptr<Board> &_board, InputOutputArrayOfArra
   */
 void refineDetectedMarkers(InputArray _image, Ptr<Board> &_board,
                            InputOutputArrayOfArrays _detectedCorners, InputOutputArray _detectedIds,
-                           InputOutputArray _rejectedCorners, InputArray _cameraMatrix,
+                           InputOutputArrayOfArrays _rejectedCorners, InputArray _cameraMatrix,
                            InputArray _distCoeffs, float minRepDistance, float errorCorrectionRate,
                            bool checkAllOrders, OutputArray _recoveredIdxs,
                            const Ptr<DetectorParameters> &_params) {

